src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    add: (number) => {
+      calculator.result += number;
+
+      return calculator;
+    },
+    subtract: (number) => {
+      calculator.result -= number;
+
+      return calculator;
+    },
+    multiply: (number) => {
+      calculator.result *= number;
+
+      return calculator;
+    },
+    divide: (number) => {
+      calculator.result /= number;
+
+      return calculator;
+    },
+    reset: () => {
+      calculator.result = 0;
+
+      return calculator;
+    },
+    operate: (method, number) => {
+      return method(number);
```

Comment:
```suggestion
      this.result = method(this.result, number);
      return this;
```

Reply:
tests don't pass after the proposed changes